internal/server/iigointernal/judiciary.go

```diff
@@ -0,0 +1,196 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+	"github.com/pkg/errors"
+)
+
+type judiciary struct {
+	JudgeID           shared.ClientID
+	budget            shared.Resources
+	presidentSalary   shared.Resources
+	BallotID          int
+	ResAllocID        int
+	speakerID         shared.ClientID
+	presidentID       shared.ClientID
+	EvaluationResults map[shared.ClientID]roles.EvaluationReturn
+	clientJudge       roles.Judge
+}
+
+func (j *judiciary) init() {
+	j.BallotID = 0
+	j.ResAllocID = 0
+}
+
+// returnPresidentSalary returns the salary to the common pool.
+func (j *judiciary) returnPresidentSalary() shared.Resources {
+	x := j.presidentSalary
+	j.presidentSalary = 0
+	return x
+}
+
+// withdrawPresidentSalary withdraws the president's salary from the common pool.
+func (j *judiciary) withdrawPresidentSalary(gameState *gamestate.GameState) error {
+	var presidentSalary = shared.Resources(rules.VariableMap["presidentSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(presidentSalary, gameState)
+	if withdrawError == nil {
+		j.presidentSalary = presidentSalary
+	}
+	return withdrawError
+}
+
+// sendPresidentSalary sends the president's salary to the president.
+func (j *judiciary) sendPresidentSalary() {
+	if j.clientJudge != nil {
+		amount := j.clientJudge.PayPresident()
+		executiveBranch.budget = amount
+		return
+	}
+	amount, _ := j.PayPresident()
+	executiveBranch.budget = amount
+}
+
+// PayPresident pays the president salary.
+func (j *judiciary) PayPresident() (shared.Resources, error) {
+	hold := j.presidentSalary
+	j.presidentSalary = 0
+	return hold, nil
+}
+
+// setSpeakerAndPresidentIDs set the speaker and president IDs.
+func (j *judiciary) setSpeakerAndPresidentIDs(speakerId shared.ClientID, presidentId shared.ClientID) {
+	j.speakerID = speakerId
+	j.presidentID = presidentId
+}
+
+// InspectHistory checks all actions that happened in the last turn and audits them.
+// This can be overridden by clients.
+func (j *judiciary) inspectHistory() (map[shared.ClientID]roles.EvaluationReturn, error) {
+	j.budget -= serviceCharge
```

Comment:
What happens if the budget is too low?

Reply:
All of costing will be overhauled soon. For MVP there is no way that the budget can become negative as each session has a fixed cost. These costs will be variable post-MVP and then a check will become necessary.